components/ruby/lib/chef-licensing/license_key_fetcher.rb

```diff
@@ -138,7 +144,13 @@ def perform_global_operations
       # Scenario: When a user is prompted for license expiry and license is not yet renewed
       if new_keys.empty? && %i{prompt_license_about_to_expire prompt_license_expired prompt_license_exhausted}.include?(config[:start_interaction])
         # Not blocking any license type in case of expiry or for commercial license exhaustion
-        return @license_keys
+        if config[:start_interaction] == :prompt_license_exhausted
+          # If user has exhausted commercial license, we are not blocking
+          # we are blocking only when user has exhausted free license, hence LicenseKeyNotFetchedError is raised for free license
+          return @license_keys if license.license_type.downcase == "commercial"
```

Comment:
Instead of multiple if statements on lines 147 and 150, we can merge them into one if condition.
` if (config[:start_interaction] == :prompt_license_exhausted) && (license.license_type.downcase == "commercial")`

Reply:
merging the condition as `if (config[:start_interaction] == :prompt_license_exhausted) && (license.license_type.downcase == "commercial")` would require another check that the config[:interaction] is not :prompt_license_exhausted as I do not want to return the license keys for free license.

Reply:
Got it. Makes sense.